test/r_c_shortest_paths_test.cpp

```diff
@@ -199,6 +199,89 @@ class dominance_spptw
+        marked.insert(v);
+    }
+    spp_spptw_marked_res_cont& operator=( const spp_spptw_marked_res_cont& other )
+    {
+        if( this == &other )
+            return *this;
+        this->~spp_spptw_marked_res_cont();
+        new( this ) spp_spptw_marked_res_cont( other );
+        return *this;
+    }
+    int cost;
+    int time;
+    std::set<SPPRC_Example_Graph::vertex_descriptor> marked;
+};
+
+bool operator==( const spp_spptw_marked_res_cont& res_cont_1,
+                 const spp_spptw_marked_res_cont& res_cont_2 )
+{
+    return res_cont_1.cost == res_cont_2.cost
+            && res_cont_1.time == res_cont_2.time
+            && res_cont_1.marked == res_cont_2.marked;
+}
+
+bool operator<( const spp_spptw_marked_res_cont& res_cont_1,
+                const spp_spptw_marked_res_cont& res_cont_2 )
+{
+    if( res_cont_1.cost > res_cont_2.cost || res_cont_1.time > res_cont_2.time) {
+        return false;
+    }
+
+    if( !std::includes( res_cont_2.marked.begin(),
+                        res_cont_2.marked.end(),
+                        res_cont_1.marked.begin(),
+                        res_cont_1.marked.end() ) ) {
+        return false;
+    }
+
+    if( res_cont_1.cost == res_cont_2.cost ) {
+        return res_cont_1.time < res_cont_2.time;
+    }
+    return true;
+}
+
+class ref_spptw_marked {
+public:
+    inline bool operator()(const SPPRC_Example_Graph &g,
+                           spp_spptw_marked_res_cont &new_cont,
+                           const spp_spptw_marked_res_cont &old_cont,
+                           graph_traits
+                                   <SPPRC_Example_Graph>::edge_descriptor ed) const {
+        const graph_traits <SPPRC_Example_Graph>::vertex_descriptor dest = target(ed, g);
+
+        if(old_cont.marked.find(dest) != old_cont.marked.end()) {
+            return false;
+        }
+
+        const SPPRC_Example_Graph_Arc_Prop& arc_prop = get( edge_bundle, g )[ed];
+        const SPPRC_Example_Graph_Vert_Prop& vert_prop = get( vertex_bundle, g )[dest];
+        new_cont.cost = old_cont.cost + arc_prop.cost;
+        new_cont.marked = old_cont.marked;
+        new_cont.marked.insert(dest);
+        int& i_time = new_cont.time;
+        i_time = old_cont.time + arc_prop.time;
+        i_time < vert_prop.eat ? i_time = vert_prop.eat : 0;
+        return i_time <= vert_prop.lat;
+}
+};
+
+class dominance_spptw_marked {
+public:
+    inline bool operator()( const spp_spptw_marked_res_cont& res_cont_1,
+                            const spp_spptw_marked_res_cont& res_cont_2 ) const {
+        return res_cont_1.time <= res_cont_2.time
+                && res_cont_1.cost <= res_cont_2.cost
+                && std::includes( res_cont_1.marked.begin(),
+                                  res_cont_1.marked.end(),
+                                  res_cont_2.marked.begin(),
+                                  res_cont_2.marked.end() );
+    }
+};
+
```

Comment:
I think some brief comments on these new classes would help the reviewers understand what they do.